etl/src/main/java/org/pih/hivmigration/etl/sql/ProgramMigrator.groovy

```diff
@@ -0,0 +1,252 @@
+            
+            insert into hivmigration_outcome(outcome, outcome_concept_id) values("DIED", @outcome_died);
+            insert into hivmigration_outcome(outcome, outcome_concept_id) values("LOST", @outcome_lost_to_followup);
+            insert into hivmigration_outcome(outcome, outcome_concept_id) values("TREATMENT_STOPPED", @outcome_treatment_stopped);
+            insert into hivmigration_outcome(outcome, outcome_concept_id) values("TRANSFERRED_OUT", @outcome_transferred_out);
+            insert into hivmigration_outcome(outcome, outcome_concept_id) values("TRANSFERRED_INTERNALLY", @outcome_transferred_out);
+        ''')
+
+        setAutoIncrement("patient_program", "(select max(patient_program_id)+1 from patient_program)")
+        
+        loadFromOracleToMySql('''
+            insert into hivmigration_programs_raw
+                (source_patient_id,
+                starting_health_center,
+                health_center,
+                health_center_transfer_date,
+                treatment_status,
+                treatment_status_date,
+                art_start_date,
+                regimen_outcome,
+                regimen_outcome_date,
+                min_visit_date,
+                max_visit_date)
+            values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+        ''', '''
+            select 	
+                d.patient_id as source_patient_id,
+                d.starting_health_center,
+                d.health_center,
+                d.health_center_transfer_date,
+                d.treatment_status,
+                d.treatment_status_date,
+                r.start_date,
+                o.regimen_outcome,
+                o.regimen_outcome_date,
+                e.min_visit_date,
+                e.max_visit_date
+            from
+                hiv_demographics_real d, hiv_art_regimen_start_date r, hiv_reg_outcome_helper_dates o,
+                (
+                    select 		patient_id, min(encounter_date) as min_visit_date, max(encounter_date) as max_visit_date 
+                    from 		hiv_encounters 
+                    where 		type in ('intake','followup','patient_contact','accompagnateur') 
+                    and			encounter_date is not null
+                    group by 	patient_id
+                ) e
+            where
+                d.TREATMENT_STATUS != 'test' 
+            and 
+                d.patient_id = e.patient_id
+            and
+                d.patient_id = r.patient_id(+)
+            and
+                d.patient_id = o.patient_id(+)
+        ''')
+
+        executeMysql("Add enrollment_date and outcome", '''
+            UPDATE hivmigration_programs_raw
+            SET
+                # Enrollment Date in program is earlier of minimum encounter and art start date
+                enrollment_date = LEAST (IFNULL(min_visit_date, art_start_date), IFNULL(art_start_date, min_visit_date)),                
+                outcome = CASE treatment_status
```

Comment:
As a continuation of my earlier comments, I would change this outcome case statement to a series of update statements like:

```
-- Died Outcome
update hivmigration_programs_raw p
set outcome_concept_id = (select concept_id from concept where uuid = '3cdd446a-26fe-102b-80cb-0017a47871b2')
where treatment_status = 'died';

-- Treatment Stopped Outcome
update hivmigration_programs_raw p
set outcome_concept_id = (select concept_id from concept where uuid = '3cdc0d7a-26fe-102b-80cb-0017a47871b2')
where treatment_status like 'treatment_stopped%' or treatment_status = 'treatment_refused';
...
```

Reply:
Seems like you don't want to make this change - we can always do so later if helpful.

This might not be a comprehensive array of outcomes.  We may need a TODO in here to review.  In the code, there is:

final_outcome_statuses = died,abandoned,lost,treatment_refused,transferred_out,treatment_stopped_side_effects,treatment_stopped_other,refused_return_to_treatment,prev_undoc_txfer_conf,prev_undoc_txfer_not_conf,moved_in_haiti_txfer_conf,moved_in_haiti_txfer_not_conf,moved_out_haiti_txfer_conf,moved_out_haiti_txfer_not_conf,traced_not_found

abandoned_refused_statuses = abandoned,treatment_refused,lost,active_no_arvs_patient_refused

These may not all be reflected in the data, but we should definitely check.